### PR TITLE
[DOC] Improve dev setup description

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

- Write section headers as imperative steps; add a table of contents that thus reads like a recipe. - The TOC is redundant to the side bar, but I think there is value in listing it again at the top of the page as a summary recipe what needs to be done.
- Put description for dev environments in sphinx-design tabs

  Tabs make it more clear that these are alternatives. Also,  they reduce the overall length of the page, making it appear less intimidating.

### CODE REVIEW

File: doc/devel/development_setup.rst [hunk 1]
```diff
@@ -23,56 +28,54 @@ if you are using 2-factor authentication.
...
-Finish the install by the following command::
+      and activated with::
 
-    pip install -e .
+        conda activate mpl-dev
 
-Whenever you plan to work on Matplotlib, remember to ``conda activate mpl-dev``
-in your shell.
+      Whenever you plan to work on Matplotlib, remember to activate the
+      environment using above command.
```
[the remark this answers is not in the document]
Comment (timhoffm) [reply doc/devel/development_setup.rst]: Sure.

File: doc/devel/development_setup.rst [hunk 2]
```diff
@@ -23,56 +28,54 @@ if you are using 2-factor authentication.
...
 
-Note that if you have mamba installed you can replace conda with mamba in
-the above command.
+      A new `conda`_ environment can be set-up with::
 
-To activate your environment::
+        conda env create -f environment.yml
 
-    conda activate mpl-dev
+      Note that if you have mamba installed you can replace conda with mamba in
```
[the remark this answers is not in the document]
Comment (timhoffm) [reply doc/devel/development_setup.rst]: Reworded.

File: doc/devel/development_setup.rst [hunk 3]
```diff
@@ -23,56 +28,54 @@ if you are using 2-factor authentication.
...
+==============================
 You should set up a dedicated environment to decouple your Matplotlib
 development from other Python and Matplotlib installations on your system.
 
-Using virtual environments
---------------------------
-
-Here we use python's virtual environment `venv`_, but you may also use others
-such as conda.
+The simplest way to do this is to use either python's virtual environment
```
[the remark this answers is not in the document]
Comment (timhoffm) [reply doc/devel/development_setup.rst]: Yes.

File: doc/devel/development_setup.rst [hunk 4]
```diff
@@ -23,56 +28,54 @@ if you are using 2-factor authentication.
...
-environment in your shell.
+        source <file folder location>/bin/activate  # Linux/macOS
+        <file folder location>\Scripts\activate.bat  # Windows cmd.exe
+        <file folder location>\Scripts\Activate.ps1  # Windows PowerShell
 
-Conda dev environment
----------------------
-After you have cloned the repo change into the matplotlib directory.
+      Whenever you plan to work on Matplotlib, remember to activate the
+      development environment in your shell.
```
[the remark this answers is not in the document]
Comment (timhoffm) [reply doc/devel/development_setup.rst]: Yes.

**Comment by noatamir** [comment]
I like the imperative as well! 

Is it nitpicking to replace repo with repository? I think it's simpler English, and more inclusive to folks who might be using translation on our documentation.

File: doc/devel/development_setup.rst [hunk 5]
```diff
@@ -15,64 +20,61 @@ You can retrieve the latest sources with the command (see
...
+        python -m venv <file folder location>
 
-Whenever you plan to work on Matplotlib, remember to activate the development
-environment in your shell.
+      and activate it with one of the following ::
 
-Conda dev environment
----------------------
-After you have cloned the repo change into the matplotlib directory.
+        source <file folder location>/bin/activate  # Linux/macOS
```
Comment (greglucas) [inline doc/devel/development_setup.rst:54]: another tab set here for the different operating systems?

**Review by greglucas — COMMENTED** [review]
I think this is a good improvement to the wording. Looks like the tab formatting isn't quite right yet so I can't see it rendered yet.

As a follow-up I _think_ that one of the other issues we were running into during the meeting is identifying which dependencies were needed for which testing situations, so a future issue could be to update the dev/test dependencies page to be a bit more explicit about which tests are used in which situation (possibly a table/column format).

**Comment by timhoffm** [comment]
@noatamir where did you find "repo" here?

**Comment by noatamir** [comment]
Looks like you removed it in your latest commit 😉

File: doc/devel/development_setup.rst [hunk 6]
```diff
@@ -23,56 +28,54 @@ if you are using 2-factor authentication.
...
----------------------
-After you have cloned the repo change into the matplotlib directory.
+      Whenever you plan to work on Matplotlib, remember to activate the
+      development environment in your shell.
 
-A new conda environment can be set-up with::
+   .. tab-item:: conda environment
 
-    conda env create -f environment.yml
+      After you have cloned the repo change into the matplotlib directory.
```
[the remark this answers is not in the document]
Comment (noatamir) [reply doc/devel/development_setup.rst]: It was here ☝️

**Review by QuLogic — APPROVED** [review]

**Pull request merged**